app/controllers/collections_controller.rb

```diff
@@ -158,20 +158,49 @@ def ensure_correct_collection_provider
-    return { error: 'Failed to retrieve collection from CMR' } unless original_collection_native_xml.success?
+    cmr_response_original = cmr_client.get_concept(params[:id],token, {})
+    unless cmr_response_original.success?
+      Rails.logger.error("Failed to retrieve concept #{params[:id]} from #{Rails.configuration.cmr_env} environment.")
+      Rails.logger.info("User #{current_user.urs_uid} attempted to retrieve collection #{@collection['EntryTitle'] || params[:id]} from #{Rails.configuration.cmr_env} environment.")
+      return { error: 'Failed to retrieve collection from CMR' }
+    end
+
+    content_type = cmr_response_original.headers.fetch('content-type').split(';')[0]
+    if content_type.include?('application/vnd.nasa.cmr.umm+json')
+      Rails.logger.info("User #{current_user.urs_uid} attempted to generate a loss report for collection #{@collection['EntryTitle'] || params[:id]} (with provider #{current_user.provider_id}) from #{Rails.configuration.umm_c_version} to #{content_type}. There is no loss report available for records that already possess UMM JSON format.")
+      return { error: 'This collection is already in UMM format so there is no loss report' }
+    end
 
-    content_type = original_collection_native_xml.headers.fetch('content-type').split(';')[0]
-    return { error: 'This collection is already in UMM format so there is no loss report' } if content_type.include?('application/vnd.nasa.cmr.umm+json')
 
-    translated_collection_native_xml = cmr_client.translate_collection(JSON.pretty_generate(@collection), "application/#{Rails.configuration.umm_c_version}; charset=utf-8", content_type,  skip_validation=true)
-    return { error: 'Failed to translate collection from UMM back to native format' } unless translated_collection_native_xml.success?
+    cmr_response_translated = cmr_client.translate_collection(JSON.pretty_generate(@collection), "application/#{Rails.configuration.umm_c_version}; charset=utf-8", content_type,  skip_validation=true)
+    unless cmr_response_translated.success?
+      Rails.logger.error("Failed to translate concept #{params[:id]} from UMM back to #{content_type}")
+      Rails.logger.info("User #{current_user.urs_uid} attempted to translate collection #{@collection['EntryTitle'] || params[:id]} (with provider #{current_user.provider_id}) from #{Rails.configuration.umm_c_version} to #{content_type} without success.")
+      return { error: "Failed to translate collection from UMM JSON back to #{content_type}" }
+    end
+
+
+    # ISO and DIF collections (in XML form) contain namespaces that cause errors when passed to loss_report_helper.rb.
+    # Specifically, when nodes are evaluated individually, (their namespace definitions remaining at the top of the xml)
+    # their prefixes are undefined in the scope of the evaluation and therefore raise errors. Removing the namespaces
+    # eliminates this issue.
+    if content_type.include?('iso') || content_type.include?('dif')
```

Comment:
I think this would be _slightly_ less repetitive and a bit clearer if you changed this to:
```
original_collection_native_xml_noko_obj = Nokogiri::XML(cmr_response_original.body) { |config| config.strict.noblanks }
      translated_collection_native_xml_noko_obj = Nokogiri::XML(cmr_response_translated.body) { |config| config.strict.noblanks }

if content_type.include?('iso') || content_type.include?('dif')
  original_collection_native_xml_noko_obj.remove_namespaces!
  translated_collection_native_xml_noko_obj.remove_namespaces!
end
```

Reply:
Yeah, this is cleaner